### PR TITLE
Add t-digest functions to Presto

### PR DESCRIPTION
**Pull request by tdcmeehan**
```
== RELEASE NOTES ==

General Changes
* Add tdigest_agg, merge(tdigest), value_at_quantile(tdigest, quantile), values_at_quantiles(tdigest, quantiles), quantile_at_value(tdigest, quantile), quantiles_at_values(tdigest, quantile) for creating, merging, and querying t-digests

SPI Changes
* isHidden attribute on AggregationFunction and ScalarFunction removed and replaced with visibility
```

### CODE REVIEW

**Comment by wenleix** [comment]
@tdcmeehan : You might need to set yourself as the commit author and add Gaston as co-author. See https://help.github.com/en/github/committing-changes-to-your-project/creating-a-commit-with-multiple-authors :)

**Comment by wenleix** [comment]
Assume @rongrong  has no further comment, @tdcmeehan can you rebase it so @rongrong or I can merge it?

**Comment by tdcmeehan** [comment]
@rongrong do you also approve `Migrate function isHidden to new visibility enum`?

File: presto-spi/src/main/java/com/facebook/presto/spi/function/SqlFunctionVisibility.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.function;
+
+public enum SqlFunctionVisibility
+{
+    PUBLIC,
+    HIDDEN,
+    /**
+     * Beta functions are marked to indicate that the function is experimental and may change in the future.  They have the same visibility as HIDDEN by default, but
```
Comment (rongrong) [inline presto-spi/src/main/java/com/facebook/presto/spi/function/SqlFunctionVisibility.java]: I'm wondering, are we going to have "alpha" functions? Is it better to name the value "experimental" as explained in the comment?

**Comment by rongrong** [comment]
> @rongrong do you also approve `Migrate function isHidden to new visibility enum`?

Yes the commit looks good in general. Please think about the naming. Whether it makes more sense to call it `EXPERIMENTAL`. Thanks for working on this! ^_^

**Pull request merged**